File: QB_Challenges/PrepInsta/JAVA/04_Sum_Between_Range/SORN_Recursion.java
import java.util.Scanner;

public class SORN_Recursion {
    
    public static int sorn(int sum, int num1, int num2) {
        if(num1 > num2){
            return sum;
        }
        return num1 + sorn(sum, num1+1, num2);
    }

    public static void main(String[] args) {
                Scanner sc = new Scanner(System.in);
        System.out.print("Enter the first number: ");
        int num1 = sc.nextInt();
        System.out.print("Enter the second number: ");
        int num2 = sc.nextInt();
        sc.close();
        int sum = sorn(0, num1, num2);
        System.out.println("The sum of "+num1+" & "+num2+" is: "+sum);

    }
}

File: QB_Challenges/PrepInsta/CPP/04_Sum_Between_Range/SORN_Recursion.cpp
#include <iostream>
using namespace std;

int SORN(int sum, int n1, int n2){
    if(n1>n2){
        return sum;
    }
    return n1 + SORN(sum, n1+1, n2);
}

int main(){
    int n1, n2;
    cout <<"Enter the first number: ";
    cin >>n1;
    cout <<"Enter the second number: ";
    cin >>n2;
    int sum = SORN(0, n1, n2);
    cout <<"The sum of numbers between "<<n1<<" & "<<n2<<" is: "<<sum <<endl;
    return 0;
}